csrc/ir/utils.cpp

```diff
@@ -1310,4 +1310,121 @@ MmaOpDetails getMmaOpDetails(
   return details;
 }
 
+// Verifies the assumptions made when evaluating a fusion containing MmaOp:
+// 1. MmaOp is preceded by a broadcast.
+// 2. The inputs to MmaOp are broadcasted as the last dim for the first operand
+// and the first dim for the second operand.
+// The inputs of MmaOp will be [M, K, 1] x [1, K, N].
+// Additionally, the inputs to the MmaOp should of the `expected_input_dtype`.
```

Comment:
For posterity, can you explain why this is required?

Reply:
There is a comment on line 1351 reasoning about the dtypes. Do you want to move it upwards?